gc/base/standard/CompactScheme.cpp

```diff
@@ -1247,6 +1247,18 @@ MM_CompactScheme::doCompact(MM_EnvironmentStandard *env, MM_MemorySubSpace *memo
 		_extensions->objectModel.postMove(env->getOmrVMThread(), deadObject);
 #endif /* defined(OMR_GC_DEFERRED_HASHCODE_INSERTION) */
 
+#if defined(OMR_ENV_DATA64)
+		if(_extensions->objectModel.isIndexable(deadObject)) {
```

Comment:
like what you did in Scavenge, you can put together this block with postMove() into an inlinde wrapper CompactScheme::postObjectMove() .

And then just to make code symmetrical, you can create preObjectMove (what would just call objectModel.preMove)